internal/nodes/crd_translator.go

```diff
@@ -198,6 +199,125 @@ func BuildL7HostRule(host, key string, vsNode AviVsEvhSniModel) {
+			if !vsNode.IsSharedVS() {
+				if !lib.IsEvhEnabled() && vsNode.IsDedicatedVS() && !vsNode.IsSecure() {
+					utils.AviLog.Debugf("key: %s, Regex and App-root are not supported for insecure SNI virtual service", key)
+				} else {
+					httpPolicyRefs := vsNode.GetHttpPolicyRefs()
+					for _, httpPolicyRef := range httpPolicyRefs {
+						var regexhppMap []AviHostPathPortPoolPG
+						var redirectPort *AviRedirectPort
+						for _, hppMap := range httpPolicyRef.HppMap {
+							if hostrule.Spec.VirtualHost.ApplicationRootPath != "" {
+								if hppMap.Path != nil && len(hppMap.Path) > 0 {
+									path := hppMap.Path[0]
+									var protocol string
+									if hppMap.SvcPort == 443 || hppMap.SvcPort == 6443 {
+										protocol = "HTTPS"
+									} else {
+										protocol = "HTTP"
+									}
+									if path == "/" {
+										redirectPort = &AviRedirectPort{
+											StatusCode:    lib.STATUS_REDIRECT,
+											Protocol:      protocol,
+											Path:          path,
+											RedirectPort:  int32(hppMap.SvcPort),
+											RedirectPath:  hostrule.Spec.VirtualHost.ApplicationRootPath[1:],
+											MatchCriteria: "EQUALS",
+										}
+										if !hostrule.Spec.VirtualHost.UseRegex {
+											hppMap.Path[0] = hostrule.Spec.VirtualHost.ApplicationRootPath
+										}
+									} else if path == hostrule.Spec.VirtualHost.ApplicationRootPath {
+										for _, childPath := range vsNode.GetPaths() {
+											if childPath == "/" {
+												redirectPort = &AviRedirectPort{
+													StatusCode:    lib.STATUS_REDIRECT,
+													Protocol:      protocol,
+													Path:          childPath,
+													RedirectPort:  int32(hppMap.SvcPort),
+													RedirectPath:  hostrule.Spec.VirtualHost.ApplicationRootPath[1:],
+													MatchCriteria: "EQUALS",
+												}
+											}
+										}
+									}
+								}
+							}
+							if hostrule.Spec.VirtualHost.UseRegex {
+								if hppMap.Path != nil && len(hppMap.Path) > 0 {
+									var regexStringGroupName string
+									path := hppMap.Path[0]
+									if hostrule.Spec.VirtualHost.ApplicationRootPath != "" && path == "/" {
+										path = hostrule.Spec.VirtualHost.ApplicationRootPath
+									}
+									regexStringGroupName = lib.GetEncodedStringGroupName(host, path)
+									kv := &models.KeyValue{
+										Key: &path,
+									}
+									key_value_sg_type := "SG_TYPE_KEYVAL"
+									hppMap.MatchCase = "INSENSITIVE"
+									hppMap.MatchCriteria = "REGEX_MATCH"
+									hppMap.MatchDecodedString = true
+
+									// If node not found, add node
+									tenant := lib.GetTenant()
+									longestMatch := true
+									regexStringGroup := &models.StringGroup{
+										TenantRef:    &tenant,
+										Type:         &key_value_sg_type,
+										LongestMatch: &longestMatch,
```

Comment:
Can we use Proto.Bool() for this? Same for other fields.

Reply:
Done